In [1]:
from tensorflow.keras.preprocessing.text import one_hot


In [2]:
# Sentences
sent = ['the glass of milk',
    'the glass of juice',
    'The cup of tea',
    'I am good boy',
    'I am a good developer',
    'understand the meaning of words',
    'I am good',]

In [4]:
# Define the vocabulary size

voc_size = 10000

In [5]:
# one hot representation
one_hot_repr = [one_hot(word,voc_size)for word in sent]
one_hot_repr

[[7087, 9266, 2029, 7569],
 [7087, 9266, 2029, 6563],
 [7087, 165, 2029, 4879],
 [7997, 4957, 732, 2198],
 [7997, 4957, 1182, 732, 1109],
 [3795, 7087, 209, 2029, 1235],
 [7997, 4957, 732]]

The words are converted into these vectors
like in size of 10000 my 7087 index in 'The'
every word represnting one index
see ' The glass of' part is same in first and second

In [10]:
# Cosine similarity is usually used in recommendation systems

# Word embedding representation

from tensorflow.keras.layers import Embedding
from tensorflow.keras.preprocessing.sequence import pad_sequences
from tensorflow.keras.models import Sequential

In [11]:
import numpy as np

In [12]:
# we want to make all sentence of same length to train,
# make max sentence length 

sent_length = 8
# add either 4 zeroes at forward or backward to make it of length 8
embedded_docs = pad_sequences(one_hot_repr, padding = 'pre', maxlen = sent_length)

In [13]:
embedded_docs

array([[   0,    0,    0,    0, 7087, 9266, 2029, 7569],
       [   0,    0,    0,    0, 7087, 9266, 2029, 6563],
       [   0,    0,    0,    0, 7087,  165, 2029, 4879],
       [   0,    0,    0,    0, 7997, 4957,  732, 2198],
       [   0,    0,    0, 7997, 4957, 1182,  732, 1109],
       [   0,    0,    0, 3795, 7087,  209, 2029, 1235],
       [   0,    0,    0,    0,    0, 7997, 4957,  732]], dtype=int32)

In [14]:
## Feature representation
dim= 10 # I want 10 features

In [18]:
model = Sequential()
model.add(Embedding(voc_size,dim,input_length = sent_length))
model.compile('adam','mse')

/Users/rishabh/ML/Projects/Churn Prediction Using ANN/avenv/lib/python3.11/site-packages/keras/src/layers/core/embedding.py:100: UserWarning: Argument `input_length` is deprecated. Just remove it.
  warnings.warn(
2026-03-02 13:59:58.617780: I metal_plugin/src/device/metal_device.cc:1154] Metal device set to: Apple M4
2026-03-02 13:59:58.617851: I metal_plugin/src/device/metal_device.cc:296] systemMemory: 16.00 GB
2026-03-02 13:59:58.617866: I metal_plugin/src/device/metal_device.cc:313] maxCacheSize: 5.92 GB
2026-03-02 13:59:58.618053: I tensorflow/core/common_runtime/pluggable_device/pluggable_device_factory.cc:305] Could not identify NUMA node of platform GPU ID 0, defaulting to 0. Your kernel may not have been built with NUMA support.
2026-03-02 13:59:58.618067: I tensorflow/core/common_runtime/pluggable_device/pluggable_device_factory.cc:271] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 0 MB memory) -> physical PluggableDevice (device: 0, name: META

In [ ]:
model.predict(embedded_docs)
# every word will be represented by 10 dimenstions

2026-03-02 14:00:49.201542: I tensorflow/core/grappler/optimizers/custom_graph_optimizer_registry.cc:117] Plugin optimizer for device_type GPU is enabled.


1/1 ━━━━━━━━━━━━━━━━━━━━ 2s 2s/step


array([[[-4.3861210e-02,  2.0041611e-02, -2.3291623e-02,  1.9282226e-02,
          4.3587279e-02, -4.0993452e-02,  1.9278470e-02, -3.3326112e-02,
          1.7208446e-02,  1.9491315e-03],
        [-4.3861210e-02,  2.0041611e-02, -2.3291623e-02,  1.9282226e-02,
          4.3587279e-02, -4.0993452e-02,  1.9278470e-02, -3.3326112e-02,
          1.7208446e-02,  1.9491315e-03],
        [-4.3861210e-02,  2.0041611e-02, -2.3291623e-02,  1.9282226e-02,
          4.3587279e-02, -4.0993452e-02,  1.9278470e-02, -3.3326112e-02,
          1.7208446e-02,  1.9491315e-03],
        [-4.3861210e-02,  2.0041611e-02, -2.3291623e-02,  1.9282226e-02,
          4.3587279e-02, -4.0993452e-02,  1.9278470e-02, -3.3326112e-02,
          1.7208446e-02,  1.9491315e-03],
        [ 7.2321519e-03,  2.7866114e-02, -2.4109555e-02, -1.3200033e-02,
         -2.6614344e-02,  9.9548325e-03,  3.1765010e-02, -3.2901395e-02,
         -1.6050935e-02,  2.6135389e-02],
        [ 4.5270015e-02,  1.4897514e-02, -9.6692890e-04,  6.

In [ ]:
model.predict(embedded_docs[0])

IndexError: tuple index out of range